### PR TITLE
jonathan-davies/refine-sign-up-form

### PR DESCRIPTION
**Pull request by 1strangequark**
I refined the UI of the signup form to be more in line with the mockups. I also had fun and added a "copy to clipboard" button.
Screen Shot 2022-02-21 at 11 53 15 PM

### CODE REVIEW

File: frontend/src/AccountCreation/SignUpProfessor.tsx [hunk 1]
```diff
@@ -24,54 +26,101 @@ const SignUpProfessor = () => {
         {error && <Alert variant={"danger"}>Failed to login: {error}</Alert>}
         <br />
         <Form onSubmit={submit}>
-          <Form.Group className="mb-3" controlId="formCourseCode">
-            <Form.Label>Your Course Code:</Form.Label>
+          <Stack direction="horizontal" className="signUpFormText">
+            <Form.Group className="mb-3" controlId="formCourseCode">
+              <Form.Label>Your Course Code:</Form.Label>
```
Comment (cindyrzheng) [inline frontend/src/AccountCreation/SignUpProfessor.tsx:31]: Just a thought- maybe it's better to not show the course code until it's been generated? 
For example, if the initial state of courseCode could be "" (empty string), and then we can add a ternary operator like 

`courseCode ? <courseCode code> : null`

File: frontend/src/AccountCreation/SignUpProfessor.tsx [hunk 2]
```diff
@@ -24,54 +26,101 @@ const SignUpProfessor = () => {
...
+            <Form.Control
+              className="signUpForm"
+              type="password"
+              name="password"
+            />
           </Form.Group>
           <Form.Group className="mb-3" controlId="formPassword">
-            <Form.Label>Re-enter Password</Form.Label>
-            <Form.Control type="password" name="password" />
+            <Form.Label className="signUpFormText">
```
Comment (cindyrzheng) [inline frontend/src/AccountCreation/SignUpProfessor.tsx:97]: https://getbootstrap.com/docs/5.0/forms/validation/ <- could be cool to add validation styles to show if passwords aren't matching!! (low priority tho just wanted to flag this if you have time but we can merge without adding this)

**Review by cindyrzheng — COMMENTED** [review]
Let me know what you think about conditionally rendering the course code! Other than that, everything looks fine and ready 2 merge :D

File: frontend/src/AccountCreation/SignUpProfessor.tsx (hunk 1 above)
Comment (1strangequark) [reply frontend/src/AccountCreation/SignUpProfessor.tsx:75]: Done. I added a spinner when the value of the course code has not been loaded.
Screen Shot 2022-02-22 at 11 56 04 AM

File: frontend/src/AccountCreation/SignUpProfessor.tsx (hunk 2 above)
Comment (1strangequark) [reply frontend/src/AccountCreation/SignUpProfessor.tsx:118]: I have form validation in my other PR

**Pull request merged**